### PR TITLE
Background tasks 10.x dev

### PR DESCRIPTION
**Pull request by alex-moreno**

### CODE REVIEW

File: README.background-tasks.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
+# ACSF Tools
+
+## Summary:
+
+Background tasks are designed to execution long running drush commands in the
```
Comment (msherron) [inline README.background-tasks.md]: 'execute', not 'execution'.

@alex-moreno - so what's the use case here? (And do you really want to identify the customer in those examples below? that seems like something you want to anonymize). I'm going to guess that this is so you can run the cohesion:sync command in parallel, right? If so - we should probably just come out and say so, as a clear use case seems like a good idea. Did we run this past the ACSF engineering team? You might want to get their blessing before we formalize this by merging it.

Comment (alex-moreno) [reply README.background-tasks.md]: thank @msherron updated that.

Yes, the product team is notified and they provided very valuable help during the process

File: README.background-tasks.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+# ############# POST DEPLOYMENT TASKS RELATED ############################
+# Custom argument. Currently used for triggering usage of post deployment tasks.
+# To trigger post deployment tasks execution using cron jobs, pass "usecronbackgroundtasks"
+custom_argument="$5"
+
+docroot="/var/www/html/$sitegroup.$env/docroot"
+DRUSH_CMD="drush9 --root=$docroot --uri=https://$domain"
+
+# Fetch if post deployment tasks are pending.
+CMD1="$(DRUSH_PATHS_CACHE_DIRECTORY="$cache_dir" $DRUSH_CMD sget cohesion_import_pending  --format=json | grep cohesion_import_pending | grep -o '.$')"
```
Comment (sdelbosc) [inline README.background-tasks.md:44]: Do we really need to mention Cohesion here?

File: composer.json [hunk 3]
```diff
@@ -21,12 +21,15 @@
   "require": {
     "php": ">=5.6.0",
     "symfony/yaml": "^3.0",
-    "composer/installers": "~1.0"
+    "composer/installers": "~1.0",
+    "ext-zip": "^1.15",
```
Comment (sdelbosc) [inline composer.json:25]: where is it used?

File: composer.json [hunk 4]
```diff
@@ -21,12 +21,15 @@
...
+    "phpmailer/phpmailer": "^6.1"
   },
   "autoload": {
     "psr-4": {
-      "Drush\\Commands\\acsf_tools\\": "."
-    }
+        "Drush\\Commands\\acsf_tools\\": "src"
```
Comment (sdelbosc) [inline composer.json:30]: Why such change?

File: composer.json [hunk 5]
```diff
@@ -38,5 +41,9 @@
         "drush.services.yml": "^9"
       }
     }
+  },
+  "require-dev": {
+      "phpunit/phpunit": "^9.1",
+      "drush/drush": "^10.2"
```
Comment (sdelbosc) [inline composer.json:47]: Sure about that?

File: scripts/background-tasks.sh [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+  DRUSH_PATHS_CACHE_DIRECTORY=$cacheDir $DRUSH_CMD cset acsf.settings site_owner_maintenance_mode $1 -y
+  site_owner_maintenance_mode="$(DRUSH_PATHS_CACHE_DIRECTORY=$cacheDir $DRUSH_CMD cget acsf.settings site_owner_maintenance_mode -y)"
+  echo "Site owner maintenance mode: $site_owner_maintenance_mode"
+}
+
+set_site_owner_maintenance_mode 0
+
+# Fetch if import is pending.
+CMD="$(DRUSH_PATHS_CACHE_DIRECTORY=$cacheDir $DRUSH_CMD sget cohesion_import_pending  --format=json | grep cohesion_import_pending | grep -o '.$')"
```
Comment (sdelbosc) [inline scripts/background-tasks.sh:42]: Why does this mention cohesion?

File: README.background-tasks.md (hunk 2 above)
Comment (alexku) [reply README.background-tasks.md:44]: This is an example use case. I think it works well for illustration purposes.

File: composer.json (hunk 3 above)
Comment (alexku) [reply composer.json:25]: zip is used to email an archive of log files after a batch of background tasks is complete

File: composer.json (hunk 4 above)
Comment (alexku) [reply composer.json:30]: @alex-moreno please help here

File: scripts/background-tasks.sh (hunk 6 above)
Comment (alexku) [reply scripts/background-tasks.sh:42]: This is an example use case. I think it works well for illustration purposes.

File: composer.json (hunk 5 above)
Comment (alexku) [reply composer.json:47]: @alex-moreno please help here

File: README.background-tasks.md [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+scripts/post-deployment-tasks-cron.sh
+```
+
+(As mentioned you'll need to get acsf_large_scale_cron.rb from your representative at Acquia).
+
+Create jobs pointing to the postdeployment script (post-deployment-tasks-cron.sh). For example, 6 jobs would look like this:
+
+```
+# Job 1
+flock -xn /tmp/bayerwsf_acqint_hsc_1.lck -c "/var/www/html/bayerwsf.01acqint/scripts/post-deployment-tasks-cron.sh 0 3"
```
Comment (alexku) [inline README.background-tasks.md]: This should be cleaned up

File: composer.json (hunk 4 above)
Comment (alex-moreno) [reply composer.json:30]: it is a better practice to keep all classes in the src folder following psr4 standards. To ensure it's not breaking the current acsf objects I map manually as well the current folder (next line)

File: composer.json (hunk 5 above)
Comment (alex-moreno) [reply composer.json:47]: this we did it with JK to ensure it was following the same versions in the project they are working on

Comment (jayakrishnanj) [reply composer.json:47]: Yes Drush 10.x is supported from BLT 11.x onwards and purpose of this 10.x branch.

**Pull request merged**